tests/integration/suite/placement/vnodes/vnodes.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vnodes
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	placementv1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(vnodes))
+}
+
+type vnodes struct {
+	place *placement.Placement
+}
+
+func (v *vnodes) Setup(t *testing.T) []framework.Option {
+	v.place = placement.New(t,
+		placement.WithMetadataEnabled(true),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(v.place),
+	}
+}
+
+func (v *vnodes) Run(t *testing.T, ctx context.Context) {
+	t.Run("register host without vnodes metadata (simulating daprd <1.13)", func(t *testing.T) {
+		v.place.WaitUntilRunning(t, ctx)
+
+		// Register the host, with API level 10 (pre v1.13)
+		msg := &placementv1pb.Host{
+			Name:     "myapp",
+			Port:     1234,
+			Entities: []string{"someactor"},
+			Id:       "myapp1",
+			ApiLevel: uint32(10),
+		}
+
+		// The host won't send the "dapr-expects-vnodes" metadata, simulating an older daprd version
+		placementMessageCh := v.place.RegisterHostWithMetadata(t, ctx, msg, nil)
+
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
```

Comment:
Does this need to be in an Eventually?

Reply:
Yeah, it does, because we're getting empty tables at first